src/syntax/simple-statements.ne

```diff
@@ -65,6 +69,27 @@ simplestatements_set_val_raw
 
 simplestatements_show -> kw_show ident {% x => track(x, { type: 'show', variable: asName(x[1]) }) %}
 
+# https://www.postgresql.org/docs/current/sql-set-transaction.html
+simplestatements_set_transaction_snapshot
+    -> kw_transaction kw_snapshot ident {% x => track(x, { type: 'set transaction snapshot', snapshotId: asName(x[2]) }) %}
+
+simplestatements_set_session
+    -> kw_session kw_characteristics %kw_as kw_transaction
+                    (simplestatements_begin_isol | simplestatements_begin_writ | simplestatements_begin_def):* {%
```

Comment:
1) session characteristics may (optionally) be separated by commas.
2) providing no transaction characteristic should not be valid (it is with this implementation)

i.e.
```sql
set session characteristics as transaction deferrable , isolation level read uncommitted, read write
```
AND 
```sql
set session characteristics as transaction deferrable  isolation level read uncommitted read write
```
are both valid.
 => there shouldb be some kind of `%comma:?` somewhere :) 

For instance, something like this should fix both issues
```
%kw_as kw_transaction sess_characteristics (%comma:? sess_characteristics):*
```

(please add UTs to check that the syntax with commas is OK too)